test/FilterTest.cpp

```diff
@@ -70,6 +70,42 @@ TEST(Filter, verifyPredicateIsAppliedCorrectlyOnLazyEvaluation) {
 TEST(Filter, verifyPredicateIsAppliedCorrectlyOnNonLazyEvaluation) {
   QueryExecutionContext* qec = ad_utility::testing::getQec();
   qec->getQueryTreeCache().clearAll();
+  IdTable idTable = makeIdTableFromVector({{true},
+                                           {true},
+                                           {false},
+                                           {false},
+                                           {true},
+                                           {true},
+                                           {false},
+                                           {false},
+                                           {false},
+                                           {false},
+                                           {true}},
+                                          asBool);
+
+  ValuesForTesting values{qec, std::move(idTable), {Variable{"?x"}}, false,
+                          {},  LocalVocab{},       std::nullopt,     true};
+  QueryExecutionTree subTree{
+      qec, std::make_shared<ValuesForTesting>(std::move(values))};
+  Filter filter{
+      qec,
+      std::make_shared<QueryExecutionTree>(std::move(subTree)),
+      {std::make_unique<sparqlExpression::VariableExpression>(Variable{"?x"}),
+       "Expression ?x"}};
+
+  auto result = filter.getResult(false, ComputationMode::FULLY_MATERIALIZED);
+  ASSERT_TRUE(result->isFullyMaterialized());
+
+  EXPECT_EQ(
+      result->idTable(),
+      makeIdTableFromVector({{true}, {true}, {true}, {true}, {true}}, asBool));
+}
```

Comment:
I think at one point (more operations)
We will benefit from testing that gets
an Operation (no need to hardcode the `|Filter`)
and a lazy input, and an expected (full) result, 
and checks, that all the cases (lazy input, materialized input, lazyInputButMaterializedResult) all do the right thing which in this case semantically all is the same.
Just thinki about this (can maybe also be a separate PR, but cleaning up the tests for frequently used and therefore important code is really something important.